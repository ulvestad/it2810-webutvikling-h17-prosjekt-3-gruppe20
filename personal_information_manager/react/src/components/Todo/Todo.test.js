import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import { shallow, mount } from 'enzyme';
import { expect } from 'chai';

import Todo from './Todo';
import FormControl from 'react-bootstrap/lib/FormControl';
import Form from 'react-bootstrap/lib/Form';

const jestExpect = global.expect;

it('renders without crashing', () => {
  shallow(<Todo />);
});

it('renders correctly', () => {
  const tree = renderer.create(<Todo />).toJSON();
  jestExpect(tree).toMatchSnapshot();
});

it('calls submit handler when form submitted', () => {
  const handleSubmit = sinon.spy(Todo.prototype, 'handleSubmit');
  const wrapper = mount(<Todo />);

  wrapper.find(Form).simulate('submit');

  expect(handleSubmit.calledOnce).to.equal(true);

  handleSubmit.restore();
});

it('adds new note when form submitted', () => {
  const wrapper = mount(<Todo />);
  const preventDefault = sinon.spy();

  wrapper.setState({ value: 'foo' });

  wrapper.instance().handleSubmit({ preventDefault });

  expect(wrapper.state('todos')).to.have.length(1);
});

it('does not add new note when form submitted if value is empty', () => {
  const wrapper = mount(<Todo />);
  const preventDefault = sinon.spy();

  wrapper.instance().handleSubmit({ preventDefault });

  expect(wrapper.state('todos')).to.have.length(0);
});

it('updates value in state when input changed', () => {
  const wrapper = mount(<Todo />);

  wrapper.find(FormControl).simulate('change', { target: { value: 'foo' } });

  expect(wrapper.state()).to.have.property('value', 'foo');
});
