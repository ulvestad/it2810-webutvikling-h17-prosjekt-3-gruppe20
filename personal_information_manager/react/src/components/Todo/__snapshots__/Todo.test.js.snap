// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="todo"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="mx-auto col-lg-8"
      >
        <h2>
          Todo
        </h2>
        <form
          className="form-inline"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <span
              className="rightMargin"
            >
              What to do:
            </span>
            <input
              className="todoMargin form-control"
              id="formInlineTodo"
              maxLength="75"
              onChange={[Function]}
              placeholder="Your todo here"
              type="text"
              value=""
            />
            <button
              className="btn btn-primary"
              disabled={false}
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
        <div>
          <div
            className="list-group"
          >
            <div
              className="box"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
